### ISSUE TITLE
Allow generator expressions to be used

### ISSUE THREAD

**Issue by svisser** [nvie/dictmerge#1]
This library is useful but it doesn't permit the use of generator expressions to pass the dictionaries. For example, it would be nice if I could write:

``` python
result = dictmerge(some_func(my_dict) for my_dict in source_of_dicts)
```

but this is currently not possible:

``` python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in dictmerge
ValueError: dictionary update sequence element #0 has length 1; 2 is required
```

Of course, one can write the following to fix it but it's less Pythonic:

``` python
result = dictmerge(*(some_func(my_dict) for my_dict in source_of_dicts))
```

**Comment by svisser** [comment]
Also, happy to make a pull request if you think this is a good addition.

**Comment by nvie** [comment]
I'd be happy to see a PR for this. There may be some cases where this can be ambiguous, though. If you want to take a stab at it, that'd be great.

**Comment by nvie** [comment]
Fixed in 0.2.0.
